Initial prototype of Blocks UI

stefan-j opened the pull request:
Allows drawing of text, circle, rectangle
Additional blocks for translation and changing color

The server was left intact.
An additional folder funblocks-base contains the Blockly logic (with bindings) and code generation,
The web folder contains new items in js and css that were used.
web/blockworld.js is based on a subset of the origin codeworld.js. Both blocks and the regular UI can be made to use the same codebase as codeworld.js with some minor changes.
web/blockly is a git submodule that contains Blockly.
The UI for the blocks currently uses bootstrap. We should probably change this later so that both Uis have the same look and feel.

I don't have a good way of compiling funblocks-base. For now I'm just running ghcjs against is with symlinks in web/js to point to the compiled js files.

The block based UI can be accessed from /blocks.html

P.S. I'm not sure why the scrollbars are suddenly black, they should be gray.

cdsmith commented on funblocks-base/funblocks-base.cabal:
```diff
@@ -0,0 +1,32 @@
+license:             Apache-2.0
+license-file:        LICENSE
+author:              Stefan Jacholke
+maintainer:          stefanjacholke@gmail.com
```
Please add this line:

    Copyright:  (c) 2016, The CodeWorld Authors

cdsmith commented on web/blockly:
```diff
@@ -0,0 +1 @@
+Subproject commit 04d6d119c076a42fb5777bb3f272d8256f2de499
```
You should add this in third_party, and symlink from here.

cdsmith commented on web/blocks.html:
```diff
@@ -0,0 +1,157 @@
+    crossorigin="anonymous">
+
+    <!-- Optional theme -->
+    <link rel="stylesheet" href="css/bootstrap.min.css">
```
Why do you need both the CDN and local version of this?  I'd prefer just the CDN version.

stefan-j commented:
Added copyright
Moved third party files to third_party

The local bootstrap.min.css is a additional theme over bootstrap. I don't think it has a CDN and might want to be modified later.

cdsmith commented on web/js/blocks_rts.js:
```diff
@@ -0,0 +1 @@
+../../funblocks-base/src/Main.jsexe/rts.js
```
Is this any different from the existing rts.js?  I think the RTS is always the same.
